### DISCUSSION TITLE
Do I need to compress my data before storing?

### DISCUSSION THREAD

**Discussion by Fish1** [pocketbase/pocketbase#2731] [Q&A] [answered]
Hello, I've built an application that store a lot of JSON data. Instead of storing the JSON data directly I've built middleware that will gzip the JSON and store it as a compressed file instead.

I was wondering if this is overkill? Does pocketbase or SQLite already compress data? Should I just store the JSON data directly?

![image](https://github.com/pocketbase/pocketbase/assets/13600696/b1d2a689-30a8-4342-a5ee-f81501fc3f47)

**Comment by ganigeorgiev** [comment, marked as answer]
I assume from the screenshot that by _"store a lot of JSON data"_ you mean uploaded JSON files, not the `json` field, right?

If that's the case, then No, files are stored on disk (or in S3) and there is no compression applied out of the box for the original uploaded file.

> Should I just store the JSON data directly?

I guess it depends mainly how large are the JSON files in the first place. Note that for very small files gzip may actually increase the size, which is why a lot of proxies usually have some min threshold (_technically the compression size doesn't really depend on the file size, but on the randomness of its data, but since larger files usually contains more repeatable data, for most of the time we can make the above assumption_).

Also keep in mind that compressing the json files on upload will also require the client to decompress them. This is not an issue if you are targeting only web since modern browsers will decompress the response for you, but for other platforms you may have to do that manually before accessing the json file content.

**Reply by Fish1** [reply]
Yeah, I do wanna stick with files because I want this uploaded to S3 eventually.

My smallest files are around 8k and compress to 700bytes, the largest are 60kb and compress to 1kb. The data is fairly repeatable.

And yeah, I've also built middleware around the file downloads, so it applies the proper headers to the pocketbase file api and the browser can decompress the gzip files.

Thanks for the tips!
